econml/automated_ml.py

```diff
@@ -0,0 +1,360 @@
+
+    Parameters
+    ----------
+
+    create_workspace: Boolean, optional, default False
+       If set to true, a new workspace will be created if the specified
+       workspace does not exist.
+
+    create_resource_group: Boolean, optional, default False
+       If set to true, a new resource_group will be created if the specified
+       resource_group does not exist.
+
+    workspace_region: String, optional
+       Region of workspace, only necessary if create_new is set to true and a
+       new workspace is being created.
+
+    subscription_id: String, required
+       Definition of a class that will serve as the parent class of the
+       AutomatedMLMixin. This class must inherit from _BaseDMLCateEstimator.
+
+    resource_group: String, required
+       Name of resource group of workspace to be created or set.
+
+    workspace_name: String, required
+       Name of workspace of workspace to be created or set.
+    """
+    try:
+        ws = Workspace(subscription_id=subscription_id, resource_group=resource_group,
+                       workspace_name=workspace_name, auth=auth)
+        # write the details of the workspace to a configuration file to the notebook library
+        ws.write_config()
+        print("Workspace configuration has succeeded.")
+    except ProjectSystemException:
+        if(create_workspace):
+            if(create_resource_group):
+                print("Workspace not accessible. Creating a new workspace and \
+                resource group.")
+                ws = Workspace.create(name=workspace_name,
+                                      subscription_id=subscription_id,
+                                      resource_group=resource_group,
+                                      location=workspace_region,
+                                      create_resource_group=create_resource_group,
+                                      sku='basic',
+                                      auth=auth,
+                                      exist_ok=True)
+                ws.get_details()
+            else:
+                print("Workspace not accessible. Set \
+                create_resource_group = True and run again to create a new \
+                workspace and resource group.")
+        else:
+            print("Workspace not accessible. Set create_workspace = True \
+            to create a new workspace.")
+
+
+def addAutomatedML(baseClass):
+    """
+    Add AutomatedMLMixin to specified base class.
+
+    In addition to a mixin that creates automatedML models from AutoMLConfig
+    objects, another mixin is added specifying modifications to the
+    AutoMLConfig. The business logic that this mixin runs is detailed below:
+
+    If the baseClass is of type NonParamDMLCateEstimator
+    the automatedML configuration must support sample weights for
+    correctness, and addAutomatedML will add functionality inject the
+    following models to the ``model_final_automl_config`` blacklist.
```

Comment:
Wouldn't it be better to use a whitelist in case new regressors are added in the future?